### PR TITLE
savedot with AbstractMetaGraph to include directed graphs, fix #86

### LINKED ISSUE #86: Why does savedot only support a directed graph?
It seems to do the same thing but is fixed on a MetaDiGraph: 
https://github.com/JuliaGraphs/MetaGraphs.jl/blob/56154277d0679049c3542a65d812014ef987161c/src/persistence.jl#L19

### PR DESCRIPTION
**Pull request by Affie**

### CODE REVIEW

File: src/persistence.jl [hunk 1]
```diff
@@ -16,8 +16,11 @@ end
 loadgraph(fn::AbstractString, gname::String, ::MGFormat) = loadmg(fn)
 savegraph(fn::AbstractString, g::AbstractMetaGraph) =  savemg(fn, g)
 
-function savedot(io::IO, g::MetaDiGraph)
-    write(io, "digraph G {\n")
+function savedot(io::IO, g::AbstractMetaGraph)
+
+    isa(g, MetaDiGraph) ? write(io, "digraph G {\n") :
```
Comment (sbromberger) [inline src/persistence.jl]: I would use `is_directed(g)` here. We already know it's some sort of metagraph.

File: src/persistence.jl [hunk 2]
```diff
@@ -38,7 +41,8 @@ function savedot(io::IO, g::MetaDiGraph)
     end
 
     for e in edges(g)
-        write(io, "$(src(e)) -> $(dst(e)) [ ")
+        isa(g, MetaDiGraph) ? write(io, "$(src(e)) -> $(dst(e)) [ ") :
```
Comment (sbromberger) [inline src/persistence.jl]: see above.

**Review by sbromberger — COMMENTED** [review]
This looks good. I'd rather avoid the use of reflection / `isa` - it seems like a code smell.

File: src/persistence.jl (hunk 1 above)
Comment (Affie) [reply src/persistence.jl]: Thanks, you are right.

File: src/persistence.jl [hunk 3]
```diff
@@ -38,7 +41,8 @@ function savedot(io::IO, g::MetaDiGraph)
     end
 
     for e in edges(g)
-        write(io, "$(src(e)) -> $(dst(e)) [ ")
+        is_directed(g) ? write(io, "$(src(e)) -> $(dst(e)) [ ") :
+                         write(io, "$(src(e)) -- $(dst(e)) [ ")
```
Comment (sbromberger) [inline src/persistence.jl]: Ah, one other thing. Why check this on each loop? If the graph is directed then we know what the string should be for all edges:
```
if is_directed(g)
  headerstr = "digraph"
  dash = "->
else
  headerstr = "graph"
  dash = "--"
end
...
write(io, "$header_str G {\n")
...
write(io, "$($src(e)) $dash $($dst(e)) [")
...
```

Comment (Affie) [reply src/persistence.jl]: I updated as per you suggestion.

**Review by sbromberger — APPROVED** [review]
Nice work! Thank you.

**Pull request merged**